cli/cli-client/src/main/java/org/infinispan/cli/benchmark/HttpBenchmark.java

```diff
@@ -0,0 +1,116 @@
+package org.infinispan.cli.benchmark;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.client.rest.RestCacheClient;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestEntity;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.RestURI;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.util.Util;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.0
+ **/
+@State(Scope.Thread)
+public class HttpBenchmark {
+   private static final long BIG_DELAY_NANOS = TimeUnit.DAYS.toNanos(1);
+   RestClient cm;
+   RestCacheClient cache;
+
+   @Param("http://127.0.0.1")
+   public String uri;
+
+   @Param("benchmark")
+   public String cacheName;
+
+   @Param("")
+   public String cacheTemplate;
+
+   @Param("16")
+   public int keySize;
+
+   @Param("1000")
+   public int valueSize;
+
+   @Param("1000")
+   public int keySetSize;
+
+   RestEntity value;
+   List<String> keySet;
+   AtomicInteger nextIndex;
+
+   @Setup
+   public void setup() {
+      RestURI uri = RestURI.create(this.uri);
+      RestClientConfigurationBuilder builder = uri.toConfigurationBuilder();
+      cm = RestClient.forConfiguration(builder.build());
```

Comment:
cm variable name related to cacheManager but the RestClient returns a restClient. I would rename this to avoid confusion with the HotrodBenchmark